server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -50,6 +50,7 @@ public class IndexShardRoutingTable {
     final List<ShardRouting> replicas;
     final List<ShardRouting> activeShards;
     final List<ShardRouting> assignedShards;
+    final List<ShardRouting> unpromotableShards;
```

Comment:
Should this and above be private?

Reply:
I think it worth making at least new one(s) private and leave the ones that are actually used outside for a separate prs..

Reply:
Thanks @idegtiarenko for the feedback! I managed to make them all private. Feel free to review again.